.github/workflows/main.yml

```diff
@@ -10,67 +10,82 @@ jobs:
 
     steps:
     # checkout repo
-    - uses: actions/checkout@v4
+    - name: Checkout
+      uses: actions/checkout@v4
 
-    # setup multi-arch build support
-    - name: Install Dependencies
-      run: |
-        sudo apt-get update
-        sudo apt-get --yes --no-install-recommends install binfmt-support qemu-user-static
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v3
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
 
-    # get branch / tag name
-    - name: Get Branch / Tag Name
+    - name: Get branch/tag name
       id: get_branch
       run: |
-        export BRANCH_NAME=$(if [[ ${GITHUB_REF} =~ "refs/tags/" ]]; then echo ${GITHUB_REF/refs\/tags\//}; else echo ${GITHUB_REF/refs\/heads\//}; fi)
+        export BRANCH_NAME=$(echo "${{ github.ref }}" | sed -e "s/refs\/heads\///g" -e "s/refs\/tags\///g")
         echo $BRANCH_NAME
         echo "BRANCH_NAME=${BRANCH_NAME}" >> $GITHUB_OUTPUT
 
-    # generate the image tag
-    - name: Get Image Tag
-      id: get_tag
+    - name: Set image tag
+      id: image_tag
       run: |
-        export TARGET_IMAGE_TAG=$(if [ "${{ steps.get_branch.outputs.NAME }}" = "latest" ]; then echo "latest"; else echo "${{ steps.get_branch.outputs.BRANCH_NAME }}"; fi)
-        echo $TARGET_IMAGE_TAG
-        echo "TARGET_IMAGE_TAG=${TARGET_IMAGE_TAG}" >> $GITHUB_OUTPUT
+        export IMAGE_TAG=$(if [[ "${{ steps.get_branch.outputs.BRANCH_NAME }}" =~ (latest|master|main) ]]; then echo "latest"; else echo "${{ steps.get_branch.outputs.BRANCH_NAME }}"; fi)
+        echo $IMAGE_TAG
+        echo "IMAGE_TAG=${IMAGE_TAG}" >> $GITHUB_OUTPUT
 
-    # generate the alternative image tag
-    - name: Get Alternate Tag
-      id: get_alt_tag
+    - name: Set alternate image tag
```

Comment:
What is the alternate image being used for? From what I see there is only ubuntu release.